### ISSUE TITLE
Componentize ControlGroup and maybe improve Control

### ISSUE THREAD

**Issue by cyrilchapon** [alex3165/react-mapbox-gl#526]
I think we could create a `ControlGroup` component, with some props like accepting various props, that could wrap controls. Like so

```
import {
  controlPosition,
  RotationControl,
  ZoomControl,
  ControlGroup
}, Map from 'react-mapbox-gl'

<Map>
  <ControlGroup
    y-position={controlPosition.X.TOP}
    x-position={controlPosition.Y.RIGHT}
  >
    <ZoomControl />
    <RotationControl />
  </ControlGroup>
</Map>
```

**Comment by cyrilchapon** [comment]
Css and philosophy talking, we could have a basic flex grid with 4 corners, like so :

```
|          |          |
|------|------|
|          |          |
```

... over the map. There would be the "control-groups-container", then each ControlGroup (.control-group) would stack accordingly inside the corners

**Comment by alex3165** [comment]
Hey @cyrilchapon, good idea although I would rather focus my time on making sure the core components of this library (`Layer`, `Source`, `GeoJSONLayer` and `Map`) are perfectly stable before working on such a component.

You can always build it and if you have something working well on your side, you can submit a PR and I will have a look and eventually introduce it in `react-mapbox-gl`.

I will close this ticket in the meantime as I don't want to make it a priority.

**Comment by cyrilchapon** [comment]
Sounds fair.

I'll see as I'm busy as hell 😂

Le 12 févr. 2018 à 22:06, Alexandre Rieux <notifications@github.com<mailto:notifications@github.com>> a écrit :


Closed #526<https://github.com/alex3165/react-mapbox-gl/issues/526>.
